src/record/delete/index.ts

```diff
@@ -1,18 +1,24 @@
 import type { RestAPIClientOptions } from "../../kintone/client";
 import { buildRestAPIClient } from "../../kintone/client";
-import { deleteAllRecords } from "./usecases/delete";
+import { deleteAllRecords } from "./usecases/deleteAll";
+import { deleteRecordsByFile } from "./usecases/delete";
 import { logger } from "../../utils/log";
 
 export type Options = {
   app: string;
+  filePath?: string;
 };
 
 export const run: (
   argv: RestAPIClientOptions & Options
 ) => Promise<void> = async (options) => {
   try {
-    const { app, ...restApiClientOptions } = options;
+    const { app, filePath, ...restApiClientOptions } = options;
     const apiClient = buildRestAPIClient(restApiClientOptions);
+    if (filePath) {
+      await deleteRecordsByFile(apiClient, app, filePath);
```

Comment:
IMO: It's better to separate CSV parsing from usecase.

Reply:
@mshrtsr 

You mean that It's better to parser CSV from here to get record numbers from the specified file, then use the use case to process the data, aren't you?
If so, the file name of the use case we can change to `deleteByRecordNumber`

Reply:
@hung-cybo 
This will be a bit long, but let me talk about the basic concept of why I think `ByRecordNumber` is better. :)

I think that the usecase should not have knowledge of details.
The details here mean the concrete implementation of API calls, reading file systems, and parsing CSV.
For example, we are currently passing `apiClient` to the usecase. This has two advantages.
One is that it makes it easier to change the internal implementation of `apiClient` since the usecase only depends on the interface of `apiClient`.
The second is that it makes it easier to test. We can pass the dummy instance to usecase from outside instead of creating one internally, and we can test the usecase without real network communication.
This concept is based on the Dependency Inversion Principle and Clean Architecture.

In this case, I think that the usecase only has to know the knowledge about “deleting specific records”.
Usecase doesn’t need to know where the record list is saved on the file system or how to read and parse it.
The controller must know that the record number can be read from a file.
If the usecase accepts the file path, we must create a dummy file to test it.
Tests that cause real access to file systems or networks can be slow and flaky.
Another advantage is that we can reuse deleteRecordsByRecordNumber when cli-kintone starts supporting another input format like `--ids`, because usecase doesn’t know where the record numbers come from.

Reply:
@mshrtsr 

Thank you for the detailed explanation.

Before, I just think the `usecases`  folder contains the logic for actions that we support, such as import(add, upsert), delete(delete all, delete by file)

But, now, I can understand more clearly the design of the `usecases` folder.
So, I totally agree with you.
Let me refactor it.
Thank you.